fix: QR code and add log

m1sterc001guy opened the pull request:
Replaces every `print` on the dart side with `logToFile` so that the app can be debugged on mobile.

Also fixes a bug with the QR code scanner that prevented it from detecting scanned codes.

bradleystachurski commented on lib/utils.dart:
```diff
@@ -1,3 +1,43 @@
   BigInt get toSats => this ~/ BigInt.from(1000);
 }
+
+Future<void> logToFile(String message) async {
```
:+1:

bradleystachurski commented:
Doesn't block merging, but looks like this throws on desktop

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(40)] Unhandled Exception: UnimplementedError: getExternalStoragePath() has not been implemented.
#0      PathProviderPlatform.getExternalStoragePath (package:path_provider_platform_interface/path_provider_platform_interface.dart:75:5)
#1      getExternalStorageDirectory (package:path_provider/path_provider.dart:154:40)
#2      logToFile (package:carbine/utils.dart:11:27)
```

m1sterc001guy commented:
> Doesn't block merging, but looks like this throws on desktop

Fixed. Also added some improvements to the logger. Have not solved Rust/Flutter unified log but this was necessary to debug on mobile a bit better.
